### ISSUE TITLE
Expectation ignored when test cases are run in a certain order

### ISSUE THREAD

**Issue by skliew** [freerange/mocha#298]
Found this issue using version `1.2.1`. In the code example as follows, `test_case_2` passes even when there is no call to any method named `test1`, resulting in a false positive:

```ruby
require 'test-unit'
require 'mocha/test_unit'

class SUT
  def test1
  end
end

class MyTest < Test::Unit::TestCase
  self.test_order = :defined
  def test_case_1
    SUT.any_instance.expects(:test1).once
    SUT.new.test1
  end

  def test_case_2
    Object.any_instance.expects(:test1).once
  end
end

```

Do note that when the line `SUT.new.test1` is commented out, both test cases fail. And when `test_case_2` is run before `test_case_1`, only `test_case_2` fails which should be the correct behavior.

**Comment by floehopper** [comment]
@skliew: Thanks for reporting this issue. Setting an expectation on any instance of `Object` seems like a bit of a pathological case, but at first glance I agree the behaviour doesn't sound correct. We'll try to look into it as soon as we have some spare time. Out of interest, do you have some real tests which need to do this?

**Comment by skliew** [comment]
I actually found this problem in a real code base when a test case occasionally fails. I have to admit that the test case was itself buggy, as most of the time false positives are reported.

And thanks for the great library :)

**Comment by floehopper** [comment]
@skliew Hmm. Sorry for taking so long to get back to you. I just did a bit of digging and it appears this might be a more fundamental problem with the Test::Unit integration. I'm continuing to investigate.

**Comment by floehopper** [comment]
I take back my last comment about it being a more fundamental problem. I made the mistake of looking at this before I'd had any coffee! I'm continuing to investigate.

**Comment by floehopper** [comment]
@skliew: I've opened https://github.com/freerange/mocha/pull/331 as an attempt to fix this problem. It turns out that it is indeed a bit more general than I first thought, so many thanks for reporting it!

**Comment by floehopper** [comment]
@skliew The fix in #331 was just released in v1.6.0.
